src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -296,6 +296,191 @@ EnsureModificationsCanRun(void)
 }
 
 
+/*
+ * IsNonDistributedTable returns whether the given relation ID identifies a
+ * non-distributed table such as reference table.
+ */
+bool
+IsNonDistributedTable(Oid relationId)
+{
+	if (!IsCitusTable(relationId))
+	{
+		return false;
+	}
+	CitusTableCacheEntry *tableEntry = GetCitusTableCacheEntry(relationId);
+	return IsNonDistributedTableCacheEntry(tableEntry);
+}
+
+
+/*
+ * IsReferenceTable returns whether the given relation ID identifies a reference
+ * table.
+ */
+bool
+IsReferenceTable(Oid relationId)
```

Comment:
One point is that using `IsReferenceTable (relationId)` is tempting over `IsReferenceTableCacheEntry ()` even if we already have the `tableCacheEntry` at hand. The callers are likely to forget about `IsReferenceTableCacheEntry()`

 And, this may have performance implications in certain code-paths.

What if we change the  functions to something like and get rid of `IsReferenceTableCacheEntry `?:

```C
IsReferenceTable(Oid  relationId, CitusTableCacheEntry *optionalTableEntry)
{
     /* if the optionalTableEntry  entry is not NULL,  we're good. 
       *  else we get the cache entry and do the same*/
}
```

This may not be the best API, just a quick suggestion. Do you have any ideas on  this?

(One idea could be to verify that my concern  about performance is not relevant to the  discussion at all)